### PR TITLE
gowin: Add a DS location recognition

### PR DESCRIPTION
**Pull request by yrabbit**
For differential signals it is necessary to set the position of two pins
at once: P and N.
This commit adds that capability and also adds another style of location
setting --- with the pin letter in square brackets used in vendor tools.

Signed-off-by: YRabbit <rabbit@yrabbit.cyou>

### CODE REVIEW

File: gowin/arch.cc [hunk 1]
```diff
@@ -551,17 +554,20 @@ void Arch::read_cst(std::istream &in)
...
+                    // may be it's IOx#[AB] style?
+                    Loc loc = getLoc(match_pinloc, getGridDimX(), getGridDimY());
+                    BelId bel = getBelByLocation(loc);
+                    if (bel == BelId()) {
+                        log_error("Pin %s not found (TRBL style). \n", pinline.c_str());
+                    }
+                    std::string belname = getCtx()->nameOfBel(bel);
+                    it->second->setAttr(IdString(ID_BEL), belname);
+                } else {
+                    std::cout << "plain error:[" << pinline << "]" << std::endl;
```
Comment (gatecat) [inline gowin/arch.cc:568]: this looks like debugging that can be removed?

Comment (yrabbit) [reply gowin/arch.cc]: Oops, how did I miss that?

**Comment by yrabbit** [comment]
What does error 503 mean? Has the tar run out of space on the HDD?
room

**Comment by gatecat** [comment]
Probably spurious, certainly not because of this PR.

**Pull request merged**